### PR TITLE
Support Scala Native 0.4.0

### PR DESCRIPTION
**Pull request by lolgab**
- Drop support for Scala Native 0.3
- Remove `Release` ReleaseMode
- Make `nativeTarget` `Option[String]` like upstream
- Rewrote tests to JUnit. Since Scala Native ships with JUnit and Utest
  depends on Mill itself we can't use it to test Mill.
  JUnit is the official testing framework and stresses all the features
  that were tested with Utest and Scalatest
- Testing Scala Native for Scala 2.11, 2.12 and 2.13
- Move worker again into the `0.4` folder. It was moved outside using
  `def millSourcePath = super.millSourcePath / os.up` to reuse the code
  between worker `0.3` and `0.4`. Since now we're dropping `0.3` it
  doesn't make sense anymore to do this, and we can be prepared for a
  future 0.5

### CODE REVIEW

File: scalanativelib/test/src/HelloNativeWorldTests.scala [hunk 1]
```diff
@@ -49,38 +48,33 @@ object HelloNativeWorldTests extends TestSuite {
...
-        def ivyDeps = Agg(ivy"com.lihaoyi::utest::0.7.5")
-        def testFrameworks = Seq("utest.runner.Framework")
+        override def sources = T.sources{ millSourcePath / "src" / "no-tests" }
+        def testFrameworks = Seq("com.novocode.junit.JUnitFramework")
+        override def ivyDeps = Agg(
+          ivy"org.scala-native::junit-runtime::$sNativeVersion"  
+        )
+        override def compileIvyDeps = Agg(  
+          ivy"org.scala-native:::junit-plugin:$sNativeVersion"  
+        )
```
Comment (lefou) [inline scalanativelib/test/src/HelloNativeWorldTests.scala:77]: This looks like a repetitiion of the JUnit-variant. Maybe, we can have it in a shared trait?

File: scalanativelib/api/src/GetFrameworkResult.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package mill.scalanativelib.api;
```
Comment (lefou) [inline scalanativelib/api/src/GetFrameworkResult.java:1]: This source files should be located in a directory structure reflecting the package name. The Java Standard isn't as flexible as the Scala one. Beside the fact that it just works with zinc, it won't for pure Java compiler.

Same, for all other Java files.

**Review by lefou — COMMENTED** [review]
Thanks for keeping Scala Native support up-to-date (and ahead)! I had a quick look at it. Looks good so far. I added some comments. And we still need to wait for 0.4.0 release to really test it.

File: scalanativelib/api/src/GetFrameworkResult.java (hunk 2 above)
Comment (lolgab) [reply scalanativelib/api/src/GetFrameworkResult.java]: @lefou You're right! Moved them ✅

File: scalanativelib/test/src/HelloNativeWorldTests.scala (hunk 1 above)
Comment (lolgab) [reply scalanativelib/test/src/HelloNativeWorldTests.scala]: Extracted on a separate trait ✅

**Comment by lolgab** [comment]
@lefou I tried in every way on earth to make the Scala Native JUnit framework to work on mill but I can't make it work.
I tried manually with utest and everything works fine.
If you agree I would change it back to `utest` releasing a version for Scala Native `0.4.0` from my branch to bootstrap the test. When Mill is released I can update utest and move the test to use the official one from `com.lihaoyi`.
What do you think?

**Comment by lefou** [comment]
> @lefou I tried in every way on earth to make the Scala Native JUnit framework to work on mill but I can't make it work.
> I tried with utest (after publishing an updated version locally) and everything works fine.
> If you agree I would change the test back to `utest` releasing a version for Scala Native `0.4.0` from my fork to bootstrap the test. When Mill is released I can update utest and move the test to use the official one from `com.lihaoyi`.
> What do you think?

@lolgab Makes sense. But I have no idea how long until a newer utest is released. Did you requested your changes for merge?

Let me restart Github tests first. If everything is green, we can merge to get a new snapshot release.

**Comment by sideeffffect** [comment]
Build is green :tada:

**Comment by lolgab** [comment]
I opened a PR [here](https://github.com/lihaoyi/utest/pull/229) but it is red because it needs this mill PR to work 😄.
Circular dependencies 🤷‍♂️.
If you merge this I can start updating utest, change it back and updating all the other libs that make the Scala Native ecosystem.

**Comment by lefou** [comment]
Thank you!

**Pull request merged**